src/main/webapp/app/complaints/shared/complaints-textarea-counter.component.ts

```diff
@@ -0,0 +1,19 @@
+import { Component, Input } from '@angular/core';
+import { StringCountService } from 'app/exercises/text/participate/string-count.service';
+
+@Component({
+    selector: 'jhi-complaints-textarea-counter',
+    templateUrl: './complaints-textarea-counter.component.html',
+    providers: [],
+})
+export class ComplaintsTextAreaCounterComponent {
```

Comment:
```suggestion
export class TextAreaCounterComponent {
```
Is this component really specific to complaints? I think it could potentially be re-used for various text inputs where a character limit is relevant and therefore should get a more generic name and be moved to a general/shared module.